assets/src/edit-story/components/library/panes/shared/chipGroup/index.js

```diff
@@ -0,0 +1,212 @@
+// it gets shown even when the provider is not the selectedProvider!
+const ExpandButton = styled(Button).attrs({
+  type: BUTTON_TYPES.TERTIARY,
+  size: BUTTON_SIZES.SMALL,
+  variant: BUTTON_VARIANTS.CIRCLE,
+})`
+  display: flex;
+  position: absolute;
+  bottom: -24px;
+  background: ${({ theme }) => theme.colors.bg.tertiary};
+  max-height: none;
+  width: 32px;
+  height: 32px;
+  ${({ isExpanded }) => isExpanded && 'transform: scaleY(-1);'}
+  visibility: inherit;
+  align-self: center;
+  justify-content: center;
+  align-items: center;
+`;
+
+const ChipGroup = ({ items, selectedItemId, selectItem, deselectItem }) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const sectionRef = useRef();
+  const innerContainerRef = useRef();
+
+  const itemRefs = useRef([]);
+
+  const [focusedRowOffset, setFocusedRowOffset] = useState(0);
+
+  const handleClick = useCallback(
+    (selected, id) => {
+      if (selected && id !== null) {
+        deselectItem();
+      } else {
+        setIsExpanded(false);
+        selectItem(id);
+      }
+    },
+    [deselectItem, selectItem]
+  );
+
+  const handleExpandClick = useCallback(
+    () => setIsExpanded((currentExpanded) => !currentExpanded),
+    []
+  );
+
+  useExpandAnimation({
+    sectionRef,
+    innerContainerRef,
+    isExpanded,
+    setFocusedRowOffset,
+    collapsedHeight: CHIP_COLLAPSED_FULL_HEIGHT,
+    bottomMargin: CHIP_BOTTOM_MARGIN,
+    topMargin: CHIP_TOP_MARGIN,
+  });
+
+  useHandleRowVisibility({
+    isExpanded,
+    innerContainerRef,
+    selectedItemId,
+    setFocusedRowOffset,
+    itemRefs,
+    offsetSpacing: FOCUS_BORDER_SPACING,
+  });
+
+  const hasItems = items.length > 0;
+  useRovingTabIndex({ ref: sectionRef }, [isExpanded]);
+  useKeyDownEffect(
+    sectionRef,
+    !isExpanded ? 'down' : '',
+    () => hasItems && setIsExpanded(true),
+    [isExpanded, hasItems]
+  );
+
+  const containerId = useMemo(() => `pill-group-${uuidv4()}`, []);
+  return (
+    <Section ref={sectionRef}>
+      {hasItems && (
+        <>
+          <Container
+            id={containerId}
+            isExpanded={isExpanded}
+            role="listbox"
+            aria-label={__('List of filtering options', 'web-stories')}
+          >
+            <InnerContainer
+              role="presentation"
+              ref={innerContainerRef}
+              style={{ transform: `translateY(-${focusedRowOffset}px` }}
+            >
+              {items.map((item) => {
+                const { id, label } = item;
+                const selected = id === selectedItemId;
+
+                return (
+                  <Chip
+                    key={id}
+                    role="option"
+                    ref={(el) => {
+                      itemRefs.current[id] = el;
+                    }}
+                    active={selected}
+                    aria-selected={selected}
+                    onClick={() => handleClick(selected, id)}
```

Comment:
This is still missing the `tabIndex={selected ? 0 : -1}`, I think.

Reply:
oops, just pushed this up!

Reply:
https://github.com/google/web-stories-wp/pull/6977/files#diff-9889a6072fafb132e7b41b93dea81b0138abffbc2cfd059f374c7d3679035a96R171-R185